test/e2e-wi-tests.sh

```diff
@@ -13,72 +13,67 @@
-  kubectl annotate serviceaccount ${K8S_CONTROLLER_SERVICE_ACCOUNT} iam.gke.io/gcp-service-account=${AUTHENTICATED_SERVICE_ACCOUNT} \
-    --namespace ${CONTROL_PLANE_NAMESPACE}
+# Setup resources common to all eventing tests.
+function test_setup() {
+  pubsub_setup || return 1
+  gcp_broker_setup || return 1
+  # Create private key that will be used in storage_setup
+  create_private_key_for_pubsub_service_account || return 1
+  storage_setup || return 1
+  scheduler_setup || return 1
+  echo "Sleep 2 mins to wait for all resources to setup"
+  sleep 120
+
+  # Publish test images.
+  publish_test_images
 }
 
 function control_plane_setup() {
   # When not running on Prow we need to set up a service account for managing resources.
   if (( ! IS_PROW )); then
     echo "Set up ServiceAccount used by the Control Plane"
-    # Enable iamcredentials.googleapis.com service for Workload Identity.
-    gcloud services enable iamcredentials.googleapis.com
-    gcloud iam service-accounts create ${CONTROL_PLANE_SERVICE_ACCOUNT}
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/pubsub.admin
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/pubsub.editor
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/storage.admin
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/cloudscheduler.admin
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/logging.configWriter
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/logging.privateLogViewer
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/cloudscheduler.admin
-        # Give iam.serviceAccountAdmin role to the Google service account.
-    gcloud projects add-iam-policy-binding ${E2E_PROJECT_ID} \
-      --member=serviceAccount:${CONTROL_PLANE_SERVICE_ACCOUNT}@${E2E_PROJECT_ID}.iam.gserviceaccount.com \
-      --role roles/iam.serviceAccountAdmin
+    init_control_plane_service_account ${E2E_PROJECT_ID} ${CONTROL_PLANE_SERVICE_ACCOUNT_NON_PROW}
+    local cluster_name="$(cut -d'_' -f4 <<<"$(kubectl config current-context)")"
```

Comment:
The context can be **Renamed** so we cannot rely on the name of the context.  Can use what @grantr suggested:
```
CLUSTER_NAME=${CLUSTER_NAME:-$(gcloud config get-value run/cluster)}
CLUSTER_LOCATION=${CLUSTER_LOCATION:-$(gcloud config get-value run/cluster_location)}
```

As far as I can tell this is not a regression, so we can either fix it here or create an issue to follow up.

Reply:
This is for the e2e-wi-tests. For e2e tests, it will override the kubectl context but it won't change the gcloud config. And the cluster won't get created until the e2e set up scripts get started.

Reply:
I see. But the script also works with existing clusters, in which case the kubectl context is not overwritten, correct?

Reply:
Yep. Just wanna clarify, for e2e tests with existing cluster, we don't run this e2e set up scripts. We will run the init_xxx.scripst under ./hack repo.

Reply:
Thanks for the clarificaiton!